docs/shopify_app/sessions.md

```diff
@@ -4,68 +4,203 @@ Sessions are used to make contextual API calls for either a shop (offline sessio
+⚠️ All apps must have a shop session storage, if you started from the [Ruby App Template](https://github.com/Shopify/shopify-app-template-ruby), it's already configured to have a Shop model by default.
+
+1. If you don't already have a repository to store the access tokens, run the following generator to create a shop model to store the access tokens:
 
-[Access scopes](#access-scopes)
-  * [`ShopifyApp::ShopSessionStorageWithScopes`](#shopifyappshopsessionstoragewithscopes)
-  * [``ShopifyApp::UserSessionStorageWithScopes``](#shopifyappusersessionstoragewithscopes)
-
-[Migrating from shop-based to user-based token strategy](#migrating-from-shop-based-to-user-based-token-strategy)
-
-### Shop-based token storage (offline token)
-
-Storing tokens on the store model means that any user login associated with the store will have equal access levels to whatever the original user granted the app.
 ```sh
 rails generate shopify_app:shop_model
 ```
-This will generate a shop model which will be the storage for the tokens necessary for authentication. To enable session persistance, you'll need to configure your `/initializers/shopify_app.rb` accordingly:
+
+2. Configure `config/initializers/shopify_app.rb` to enable shop session token persistance:
 
 ```ruby
 config.shop_session_repository = 'Shop'
 ```
 
-### User-based token storage (online token)
+##### User (online) token storage
+If your app has user interactions and would like to control permission based on individual users, you need to configure a User token storage to persist unique tokens for each user.
+
+[Shop (offline) tokens must still be maintained](#shop-offline-token-storage).
 
-A more granular control over the level of access per user on an app might be necessary, to which the shop-based token strategy is not sufficient. Shopify supports a user-based token storage strategy where a unique token to each user can be managed. Shop tokens must still be maintained if you are running background jobs so that you can make use of them when necessary.
+1. Run the following generator to create a user model to store the individual based access tokens
 ```sh
-rails generate shopify_app:shop_model
 rails generate shopify_app:user_model
 ```
 
-This will generate a user and shop model which will be the storage for the tokens necessary for authentication. To enable session persistance, you'll need to configure your `/initializers/shopify_app.rb` accordingly:
+2. Configure `config/initializers/shopify_app.rb` to enable user session token persistance:
 
 ```ruby
-config.shop_session_repository = 'Shop'
 config.user_session_repository = 'User'
 ```
 
 The current Shopify user will be stored in the rails session at `session[:shopify_user]`
 
-Read more about Online vs. Offline access [here](https://shopify.dev/apps/auth/oauth/access-modes).
+##### In-memory Session Storage for testing
+The `ShopifyApp` gem includes methods for in-memory storage for both shop and user sessions.
```

Comment:
```suggestion
The `ShopifyApp` gem includes methods for in-memory storage for both shop and user sessions. In-memory storage is intended to be used by Rails apps that don't utilize a database. 
```

Reply:
I think a better way would be to let them know it should only be used for testing purposes since it's not available in production environments??

https://github.com/Shopify/shopify_app/blob/df4d24660b9b0fabfca0fa42bb6e7706e9b381a6/lib/shopify_app/session/in_memory_session_store.rb#L22-L25